packages/react-router/modules/Route.js

```diff
@@ -29,37 +29,22 @@ class Route extends React.Component {
     location: PropTypes.object
   }
 
-  static childContextTypes = {
-    router: PropTypes.object.isRequired
+  state = {
+    match: this.computeMatch(this.props)
   }
 
   getChildContext() {
     return {
-      router: this.router
+      router: {
+        ...this.context.router,
+        match: this.state.match
+      }
     }
   }
 
-  componentWillMount() {
-    const parentRouter = this.context.router
-
-    this.router = {
```

Comment:
If we're going to remove `this.router`, we should at least have a few tests that ensure correct behavior. Also, Redux users are only 1 group that uses `shouldComponentUpdate`. Anyone else who's using it may still benefit from us mutating `context.router`.